jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -26,16 +27,17 @@
 import com.microsoft.jdtls.ext.core.model.PackageNode;
 
 public final class ProjectCommand {
-    public static final String DEFAULT_PROJECT_NAME = "jdt.ls-java-project";
 
     public static List<PackageNode> execute(List<Object> arguments, IProgressMonitor monitor) {
         String workspaceUri = (String) arguments.get(0);
+        IPath workspacePath = ResourceUtils.canonicalFilePathFromURI(workspaceUri);
+        String invisibleProjectName = getWorkspaceInvisibleProjectName(workspacePath);
 
         IProject[] projects = getWorkspaceRoot().getProjects();
         ArrayList<PackageNode> children = new ArrayList<>();
-        List<IPath> paths = Arrays.asList(ResourceUtils.filePathFromURI(workspaceUri));
+        List<IPath> paths = Arrays.asList(workspacePath);
         for (IProject project : projects) {
-            if (project.exists() && !DEFAULT_PROJECT_NAME.equals(project.getName()) ) {
+            if (project.exists() && (ResourceUtils.isContainedIn(project.getLocation(), paths) || Objects.equals(project.getName(), invisibleProjectName))) {
```

Comment:
Objects.equals is reference equals?

Reply:
it's an utility method which guard the null case gracefully.